### PR TITLE
Jiterpreter opcode updates + implement safepoints

### PR DESCRIPTION
**Pull request by kg**
This PR implements some of the new interpreter opcodes (CPOBJ_VT_NOREF and STOBJ_VT_NOREF) while also adding support for doing safepoints without leaving traces. Many safepoint branches don't go backwards (this surprised me!) so we can potentially avoid a bailout if we do the safepoint from inside.

A quick set of test runs shows that this PR reduces the time to run System.Runtime.Tests in interpreter+jiterpreter mode from 141.6sec to 137.3sec. (Note that the jiterpreter is still not enabled by default)

### CODE REVIEW

File: src/mono/wasm/runtime/jiterpreter.ts [hunk 1]
```diff
@@ -2925,6 +2949,13 @@ function append_bailout (builder: WasmBuilder, ip: MintOpcodePtr, reason: Bailou
     builder.appendU8(WasmOpcode.return_);
 }
 
+function append_safepoint (builder: WasmBuilder, ip: MintOpcodePtr) {
+    builder.local("frame");
+    // Not ip_const, because we can't pass relative IP to do_safepoint
+    builder.i32_const(ip);
+    builder.callImport("safepoint");
```
Comment (vargaz) [inline src/mono/wasm/runtime/jiterpreter.ts:2956]: Would be nice to inline the if (G_UNLIKELY (...)) part of the safepoint code here, its almost always false.

**Review by vargaz — APPROVED** [review]

File: src/mono/wasm/runtime/jiterpreter.ts (hunk 1 above)
Comment (kg) [reply src/mono/wasm/runtime/jiterpreter.ts:2956]: Is there some way for us to export the flag as a wasm global? If so, I could grab it out of the module and import it.

Comment (vargaz) [reply src/mono/wasm/runtime/jiterpreter.ts:2956]: It has an address in linear memory, so just taking that address and loading from it should work.

Comment (kg) [reply src/mono/wasm/runtime/jiterpreter.ts:2956]: Good point, that's probably only slightly slower, and I can just hand the address out. Thanks.

**Pull request merged**